File: src/main/java/com/leet/easy/MaximumSubarray.go
package main

func maxSubArray(nums []int) int {
	running, ans := nums[0], nums[0]
	for i := 1; i < len(nums); i++ {
		num := nums[i]
		running += num
		if running < num {
			running = num
		}
		if running > ans {
			ans = running
		}
	}
	return ans
}

File: src/main/java/com/leet/easy/MaximumSubarray.exs
defmodule Solution do
  @spec max_sub_array(nums :: [integer]) :: integer
  def max_sub_array(nums) do
    [head | tail] = nums

    {_, ans} =
      Enum.reduce(tail, {head, head}, fn elem, {sum, ans} ->
        if sum + elem < elem, do: {elem, max(ans, elem)}, else: {elem + sum, max(ans, elem + sum)}
      end)

    ans
  end
end
